### PR TITLE
My first pull request

### PR DESCRIPTION
**Pull request by FrederickMih**
## Enumerable methods:
-My Project on Enumerable methods

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
# Status: Invalid

**Hello team, I marked this project as invalid because I could not find the spec file with tests for your enumerable methods. Please see the project specifications [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/).**

## Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

**Review by LamiaSristy — CHANGES_REQUESTED** [review]
# Status: Change Required. 🔴 

Hello @FrederickMih  🙌
Great work so far. 👏 👏 

However, we still need to implement some changes to get the project approved!🛠️

Please check my points here: 👇

`[REQUIRED]`

- [ ]  Please correct the setup section in your README file. Currently, It has the wrong pieces of information. See the screenshot below: 
README

- [ ]  Your `my_all`, `my-any?` and `my_none` method should check for Regexp and Numeric Class. Write logic for checking the followings:👇

       ```
	   # my_all?
	   puts %w[ant bear cat].my_all?(/t/) #=> false
       puts [1, 2i, 3.14].my_all?(Numeric) #=> true
       puts [].my_all? #=> true
	   
	   #.my_any?
	   puts %w[ant bear cat].my_any?(/d/) #=> false
       puts [nil, true, 99].my_any?(Integer) #=> true
       puts [nil, true, 99].my_any? #=> true
       puts [].my_any? #=> false
	   
	   #my_none?
	   puts %w[ant bear cat].my_none?(/d/) #=> true
	   puts [1, 3.14, 42].my_none?(Float) #=> false
	   puts [].my_none? #=> true
	   puts [nil].my_none? #=> true
	   puts [nil, false].my_none? #=> true
	   puts [nil, false, true].my_none? #=> false
	   
	   
	   ```
- [ ] Your `my_inject` method is not working for ranges. Write logic for checking the followings:👇
     ```
	 puts ((1..5).my_inject { |sum, n| sum + n }) #=> 15

Please make the changes and ask for a re-review.

Feel free to reach me at @lamiasristy 🙋

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+end
+# rubocop:enable Metrics/ModuleLength
+# rubocop:enable Metrics/CyclomaticComplexity
+# rubocop:enable Metrics/PerceivedComplexity
+
+
+# def multiply_els(arr)
+# arr.my_inject { |result, elem| result * elem }
+# end
+# puts multiply_els [2, 4, 5]
```
Comment (oloomoses) [inline enumerable.rb]: - [ ] Remove the commented code from your file. They make your project look messy

**Review by oloomoses — COMMENTED** [review]
# Status: Request Changes
Hello @FrederickMih,
You are making good progress on this project :muscle: 
However, some changes need to be done before submitting another review:
- [ ] The README file still has some misleading pieces of information. Kindly update it.
- [ ] Ensure that there are no commented pieces of code in your file. They make your work look messy
- [ ] Ensure that you add a descriptive pull request

### Also ensure that
#### my_each 
- [ ] returns an Enumerator if no block is given
- [ ] when a block is given when self is a range returns the range itself after calling the given block once for each element in self
#### my_each_with_index
- [ ] returns an enumerator if no block is given
- [ ] when a block is given when self is a range returns the range itself after calling the given block once for each element in self
#### my_select, my_map
- [ ] returns an Enumerator if no block is given

#### my_count 
- [ ] returns the number of items in enum through enumeration::range

- [ ] Implement `multiply_els` using `my_inject`
### You can use the [ruby](https://ruby-doc.org/core-2.6.5/Enumerable.html) for reference.

## Feel free to reach me on slack [Oloo Moses] should you have any doubts or concerns
# [Oloo Moses]

**Review by Georjane — APPROVED** [review]
## Status: PR Approved ✔️ ✔️
### Hi Team 🙌
Congratulations on successfully completing the Enumerable project. Your efforts can be clearly seen from your work. 
It’s a thumbs up for me. 👍 So you can go ahead and merge this PR. 

Goodluck on your next project and Happy Coding 💻

**Comment by FrederickMih** [comment]
This project required me to implement my own version of Ruby’s enumerable methods

**Pull request merged**